frontend/src/metabase/common/components/FilterPicker/NumberFilterPicker.tsx

```diff
@@ -1,31 +1,215 @@
 import { t } from "ttag";
-import { Box, Button, Flex, Text } from "metabase/ui";
+import { useState, useMemo } from "react";
+import { Box, Button, Flex, Text, NumberInput } from "metabase/ui";
 import * as Lib from "metabase-lib";
+
+import FieldValuesWidget from "metabase/components/FieldValuesWidget";
+import Field from "metabase-lib/metadata/Field";
 import type { FilterPickerWidgetProps } from "./types";
 import { BackButton } from "./BackButton";
 import { Header } from "./Header";
 import { Footer } from "./Footer";
 
+import { FilterOperatorPicker } from "./FilterOperatorPicker";
+
+const numberValueTypes = ["single", "multiple", "two", "none"];
+type NumberValueType = typeof numberValueTypes[number];
+
+const operatorTypes: Record<NumberValueType, Lib.FilterOperatorName[]> = {
+  single: [">", ">=", "<", "<="],
+  multiple: ["=", "!="],
+  two: ["between"],
+  none: ["is-null", "not-null", "is-empty", "not-empty"],
+};
+
 export function NumberFilterPicker({
   query,
   stageIndex,
   column,
   filter,
   onBack,
+  onChange,
+  onClose,
 }: FilterPickerWidgetProps) {
   const columnName = Lib.displayInfo(query, stageIndex, column).longDisplayName;
+  const filterParts = filter
+    ? Lib.numberFilterParts(query, stageIndex, filter)
+    : null;
+
+  const [operatorName, setOperatorName] =
```

Comment:
I opted to use the operatorName instead of the opaque operator here because it makes it so we don't have to call `displayInfo` all over the place to handle all the special UI conditions. We simply convert it back to an opaque operator before updating the filter.

(this feels a little silly because `numberFilterClause` then turns the opaque operator back into an operatorName.

Reply:
Yeah the idea is to force obtaining `FilterOperator` via a MLv2 function. There are some edge cases with available operators for columns (e.g. PKs, FKs have a subset of operators) that we shouldn't case about. The way I see it now is that there is a list of operators MLv2 returns (via `FilterOperator`) and the list of operators that the UI widget supports, and we intersect these lists.